src/researches/largestKeywordDistance.js

```diff
@@ -0,0 +1,70 @@
+const getIndicesByWord = require( "../stringProcessing/indices" ).getIndicesByWord;
+const forEach = require( "lodash/forEach" );
+const sortBy = require( "lodash/sortBy" );
+const normalizeQuotes = require( "../stringProcessing/quotes.js" ).normalize;
+
+/**
+ * Calculates the largest percentage of the text without a keyword.
+ *
+ * @param {Paper} paper The paper to check the keyword distance for.
+ * @returns {number} Returns the largest percentage of the text between two keyword occurrences
+ *                   or a keyword occurrence and the start/end of the text.
+ */
+module.exports = function( paper ) {
+	let text = paper.getText();
+	text = text.toLowerCase();
+	text = normalizeQuotes( text );
+	const textLength = text.length;
+	let keyword = paper.getKeyword();
+	keyword = keyword.toLowerCase();
+	keyword = normalizeQuotes( keyword );
+	const keywordIndices = getIndicesByWord( keyword, text );
+	let keywordDistances = [];
+
+	/*
+	 * Find the distance (in terms of percentage of the text) between two keywords,
+	 * between the beginning of the text and the first keyword,
+	 * and the last keyword and the end of the text.
+	 */
+	forEach( keywordIndices, function( keywordIndex ) {
```

Comment:
I'd much rather see this as a separate function. Also please ensure the docblock is according to the documentation standards.

Additionally, I'd also prefer seeing whatever comes from this `forEach` to be returned ASAP, making the usage of `else if` and `else` obsolete.